### PR TITLE
WIP : Added ng-bootstrap 

### PR DESCRIPTION
**Pull request by sendilkumarn**
Wired ng-bootstrap

Tested Navbar with ng-bootstrap elements

There is a problem with adding `NgbModules` in `app.ng2modules.ts` as a whole after this is fixed [angular-2] (https://github.com/angular/angular/issues/11280), we can do that.

### CODE REVIEW

**Comment by deepu105** [comment]
Sendil, lets do ng-bootstrap first.  We can do a separate PR for bs4 else
it will too much to review and merge. Create another ticket for that and
tag it with ng2 label

Thanks & regards,
Deepu

**Comment by sendilkumarn** [comment]
Okay then. 

Let me know if this is okay. If so will start migrating other components

**Comment by deepu105** [comment]
Also remove ui-bootstrap from client 2 so that we know exactly if something
else needs to be changed.

Thanks & regards,
Deepu

On 9 Sep 2016 13:48, d4udts@gmail.com wrote:

Sendil, lets do ng-bootstrap first.  We can do a separate PR for bs4 else
it will too much to review and merge. Create another ticket for that and
tag it with ng2 label

Thanks & regards,
Deepu

On 9 Sep 2016 13:36, "Sendil Kumar N" <notifications@github.com> wrote:

Wired and works.

Navbar is working with ng-bootstrap elements

There is a problem with adding NgbModules as a whole after this is fixed
angular-2 <https://github.com/angular/angular/issues/11280>

   - Add bootstrap 4
   - Replace ui-bootstrap with ng-bootstrap elements
   - Revamp application to follow bootstrap 4 completely ( @deepu105
   <https://github.com/deepu105> we can do revamp here )
   - Remove ui-bootstrap dependancy
   - Remove bootstrap 3 dependancy
   - Polish and Publish

------------------------------
You can view, comment on, or merge this pull request online at:

  https://github.com/jhipster/generator-jhipster/pull/4109
Commit Summary

   - feat(ng2): added gulp to move ng-bootstrap files

File Changes

   - *M* generators/client-2/templates/gulp/_copy.js
   <https://github.com/jhipster/generator-jhipster/pull/4109/files#diff-0>
   (1)

Patch Links:

   - https://github.com/jhipster/generator-jhipster/pull/4109.patch
   - https://github.com/jhipster/generator-jhipster/pull/4109.diff

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/jhipster/generator-jhipster/pull/4109>, or mute the
thread

File: generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts [hunk 1]
```diff
@@ -23,6 +27,8 @@ import { NavbarComponent } from './layouts/navbar/navbar.component';
         HttpModule,
         <%=angular2AppName%>CommonModule,
         <%=angular2AppName%>AdminModule,
+        NgbCollapseModule,
```
Comment (deepu105) [inline generators/client-2/templates/src/main/webapp/app/_app.ng2module.ts]: Can you put these before our modules, after the HttpModule?

**Comment by deepu105** [comment]
@sendilkumarn i have only one minor comment, once thats taken care of you can merge this. And start migrating other ui-bs stuff to ng-bs if you have time

**Pull request merged**